packages/components/src/tab-panel/index.tsx

```diff
@@ -106,10 +105,25 @@ export function TabPanel( {
 	const selectedId = `${ instanceId }-${ selectedTab?.name ?? 'none' }`;
 
 	useEffect( () => {
-		if ( ! selectedTab?.name && tabs.length > 0 ) {
-			handleTabSelection( initialTabName || tabs[ 0 ].name );
+		const firstEnabledTab = tabs.find( ( tab ) => ! tab.disabled );
+		const initialTab = tabs.find( ( tab ) => tab.name === initialTabName );
+		if ( ! selectedTab?.name && firstEnabledTab ) {
+			handleTabSelection(
+				initialTab && ! initialTab.disabled
+					? initialTab.name
+					: firstEnabledTab.name
+			);
 		}
-	}, [ tabs, selectedTab?.name, initialTabName, handleTabSelection ] );
+		if ( selectedTab?.disabled && firstEnabledTab ) {
```

Comment:
Nitty, but — right now, the code makes it look like the `handleTabSelection` function can be called twice during the same effect.

Should we rewrite the logic slightly (e.g. `else if`) to make it clear that these two code branches should be exclusive?

Reply:
Sure!  Done in https://github.com/WordPress/gutenberg/pull/46471/commits/c64730cc15148882be85a50c5536de3261138152.